checker/src/main/java/org/checkerframework/checker/resourceleak/ResourceLeakAnnotatedTypeFactory.java

```diff
@@ -0,0 +1,367 @@
+  }
+
+  /**
+   * Returns the {@link MustCall#value} element/argument of the @MustCall annotation on the class
+   * type of {@code element}.
+   *
+   * <p>If possible, prefer {@link #getMustCallValue(Tree)}, which accounts for flow-sensitive
+   * refinement.
+   *
+   * @param element an element
+   * @return the strings in its must-call type
+   */
+  /* package-private */ List<String> getMustCallValue(Element element) {
+    MustCallAnnotatedTypeFactory mustCallAnnotatedTypeFactory =
+        getTypeFactoryOfSubchecker(MustCallChecker.class);
+    AnnotatedTypeMirror mustCallAnnotatedType =
+        mustCallAnnotatedTypeFactory.getAnnotatedType(element);
+    AnnotationMirror mustCallAnnotation = mustCallAnnotatedType.getAnnotation(MustCall.class);
+
+    return getMustCallValues(mustCallAnnotation);
+  }
+
+  /**
+   * Helper method for getting the must-call values from a must-call annotation.
+   *
+   * @param mustCallAnnotation a {@link MustCall} annotation, or null
+   * @return the strings in mustCallAnnotation's value element, or the empty list if
+   *     mustCallAnnotation is null
+   */
+  private List<String> getMustCallValues(@Nullable AnnotationMirror mustCallAnnotation) {
+    if (mustCallAnnotation == null) {
+      return Collections.emptyList();
+    }
+    return AnnotationUtils.getElementValueArray(
+        mustCallAnnotation, mustCallValueElement, String.class);
+  }
+
+  /**
+   * Helper method to get the temporary variable that represents the given node, if one exists.
+   *
+   * @param node a node
+   * @return the tempvar for node's expression, or null if one does not exist
+   */
+  /* package-private */
+  @Nullable LocalVariableNode getTempVarForNode(Node node) {
+    return tempVarToTree.inverse().get(node.getTree());
+  }
+
+  /**
+   * Is the given node a temporary variable?
+   *
+   * @param node a node
+   * @return true iff the given node is a temporary variable
+   */
+  /* package-private */ boolean isTempVar(Node node) {
+    return tempVarToTree.containsKey(node);
+  }
+
+  /**
+   * Registers a temporary variable by adding it to this type factory's tempvar map.
+   *
+   * @param tmpVar a temporary variable
+   * @param tree the tree of the expression the tempvar represents
+   */
+  /* package-private */ void addTempVar(LocalVariableNode tmpVar, Tree tree) {
+    tempVarToTree.put(tmpVar, tree);
+  }
+
+  /**
+   * Returns true if the type of the tree includes a must-call annotation. Note that this method may
+   * not consider dataflow, and is only safe to use on declarations, such as method trees or
+   * parameter trees. Use {@link #getMustCallValue(ImmutableSet, CFStore)} (and check for emptiness)
+   * if you are trying to determine whether a local variable has must-call obligations.
+   *
+   * @param declTree a tree representing a declaration
+   * @return whether that declaration has must-call obligations
+   */
+  /* package-private */ boolean hasMustCall(Tree declTree) {
```

Comment:
Should this be named `declarationHasMustCall`?  Should it do an `assert` on the tree kind?